snekbox/memfs.py

```diff
@@ -0,0 +1,154 @@
+"""Memory filesystem for snekbox."""
+from __future__ import annotations
+
+import logging
+import os
+import subprocess
+from collections.abc import Generator
+from contextlib import contextmanager
+from pathlib import Path
+from shutil import rmtree
+from threading import BoundedSemaphore
+from types import TracebackType
+from typing import Type
+from uuid import uuid4
+
+from snekbox.snekio import FileAttachment
+
+log = logging.getLogger(__name__)
+
+PID = os.getpid()
+
+NAMESPACE_DIR = Path("/memfs")
+NAMESPACE_DIR.mkdir(exist_ok=True)
+NAMESPACE_DIR.chmod(0o711)  # Execute only access for other users
+
+
+def mount_tmpfs(name: str, size: int | str) -> Path:
+    """Create and mount a tmpfs directory."""
+    tmp = NAMESPACE_DIR / name
+    tmp.mkdir()
+    tmp.chmod(0o711)
```

Comment:
Why is a chmod needed here? (also all the other chmods)

Reply:
I think this was a remanent when there used to be exposed access to this folder. Removed in a0c908a. The chmods for the home directory would have to stay because that needs to be read/write.